secretary.py

```diff
@@ -306,20 +306,30 @@ def _prepare_template_tags(self, xml_document):
             parent.removeChild(discard)
 
 
-    def _unescape_entities(self, xml_text):
-        # unescape XML entities gt and lt
+    @staticmethod
+    def _unescape_entities(xml_text):
+        """
+        Strips tags of the form <text:span ...> from inside Jinja elements
+        and unescapes HTML codes for >, <, & and "
+        """
         unescape_rules = {
+            r'(?is)({[{|%].*)(<.?text:s.*?>)(.*[%|}]})': r'\1\3',
+
             r'(?is)({[{|%].*)(&gt;)(.*[%|}]})': r'\1>\3',
             r'(?is)({[{|%].*)(&lt;)(.*[%|}]})': r'\1<\3',
-            r'(?is)({[{|%].*)(<.?text:s.?>)(.*[%|}]})': r'\1 \3',
+            r'(?is)({[{|%].*)(&amp;)(.*[%|}]})': r'\1&\3',
+            r'(?is)({[{|%].*)(&quot;)(.*[%|}]})': r'\1"\3'
         }
 
         for p, r in unescape_rules.items():
-            xml_text = re.sub(p, r, xml_text)
+            count = None
+            while count != 0:
+                xml_text, count = re.subn(p, r, xml_text)
```

Comment:
Using the while loop ensures all the special characters get converted which allows us to deal with expressions that may have multiple comparisons and pairs of double quotes in them such as `{% if x > 0 and x < 10 and y != "Foo" %}`.